compiler/cpp/src/thrift/generate/t_java_generator.cc

```diff
@@ -485,11 +484,6 @@ string t_java_generator::java_package() {
   return "";
 }
 
-string t_java_generator::java_suppressions() {
-  return "@SuppressWarnings({\"cast\", \"rawtypes\", \"serial\", \"unchecked\", \"unused\"})"
-         + endl;
-}
```

Comment:
How certain are you that these aren't used? I'm pretty sure some of them are still applicable in at least some situations.

Reply:
> How certain are you that these aren't used? I'm pretty sure some of them are still applicable in at least some situations.

yeah you are right - i tried to localize the warning suppression only to find it's not worth it.

Reply:
Some of these warnings should be fixed, though with better generated code. Like... why are we generating a logger that is unused? Or why are we using raw generics instead of the specific generic types when we know them? Or, why would we need to suppress serial? Why not just use the generated serialUid or a 1-up counter for custom Exceptions?

All this would take work, but eventually, we could probably get rid of these warnings suppressions by improving the generator in other parts.